src/main/java/edu/hawaii/its/api/service/AnnouncementsService.java

```diff
@@ -0,0 +1,43 @@
+package edu.hawaii.its.api.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import edu.hawaii.its.api.type.Announcement;
+import edu.hawaii.its.api.type.Announcements;
+import edu.hawaii.its.api.util.JsonUtil;
+import edu.hawaii.its.api.wrapper.AttributesResult;
+import edu.hawaii.its.api.wrapper.FindAttributesResults;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AnnouncementsService {
+    public static final Log log = LogFactory.getLog(AnnouncementsService.class);
+
+    @Value("${groupings.api.announcements}")
+    private String ANNOUNCEMENTS_ATTR_NAME;
+
+    @Value("${groupings.api.propertystring}")
+    private String ANNOUNCEMENTS_ATTR_DEF;
+
+    @Autowired
+    private GrouperApiService grouperApiService;
+
+    public Announcements setAnnouncements(AttributesResult attributesResult) {
+        //hydrate announcements by creating a list of Announcement.class objects
+        List<Announcement> announcementsList = new ArrayList<>();
+        announcementsList = JsonUtil.asList(attributesResult.getDescription(), Announcement.class); //seems to be initializing announcement
```

Comment:
These two comments are not needed either, the code under is it self-explanatory.